source-record.c

```diff
@@ -227,6 +228,15 @@ static void source_record_filter_offscreen_render(void *data, uint32_t cx,
 	UNUSED_PARAMETER(cx);
 	UNUSED_PARAMETER(cy);
 	struct source_record_filter_context *filter = data;
+
+	uint64_t frame_time_ns = obs_get_video_frame_time();
+	int count = (int)((frame_time_ns - filter->last_frame_time_ns) /
+			  obs_get_frame_interval_ns());
+	filter->last_frame_time_ns = frame_time_ns;
```

Comment:
I put this line before other return conditions so that the `count` should not jump when starting to record.